src/api.js

```diff
@@ -608,6 +608,7 @@ api.declare({
 }, async function(req, res) {
   var taskId  = req.params.taskId;
   var taskDef = req.body;
+  var clientId = await req.clientId();
 
   // Patch default values and validate timestamps
   var detail = patchAndValidateTaskDef(taskId, taskDef);
```

Comment:
Right around here would be a good place to add a check of the supplied clientId -- if supplied, it must equal `clientId`, otherwise return `res.reportError('InputError', 'If supplied, clientId must be the clientId used to make the API call', {})`.